### PR TITLE
U/fjammes/dm 335

### PR DESCRIPTION
**Pull request by jbecla**

### CODE REVIEW

File: core/modules/parser/parseTreeUtil.cc [hunk 1]
```diff
@@ -45,12 +45,12 @@ class DigraphVisitor {
     typedef std::map<AnAst, std::string> AstMap;
 
     DigraphVisitor() : i(0) {}
-    void operator()(AnAst a, C& p) {
+    void operator()(AnAst const &a, C const & p) {
```
Comment (andy-slac) [inline core/modules/parser/parseTreeUtil.cc]: Somewhat inconsistent whitespace use in reference declarations. I looked quickly at qserv code, most frequently references are declared as `Class& ref` or `Class const& ref`, very rarely as `Class &ref`. `Class const & ref` hass one additional space. I think it's better to stick to most common use.

File: core/modules/parser/parseTreeUtil.h [hunk 2]
```diff
@@ -34,7 +34,6 @@
 // System headers
 #include <cassert>
 #include <iostream>
-#include <list>
```
Comment (andy-slac) [inline core/modules/parser/parseTreeUtil.h]: vector header should be added, vector class is used in this header file.

File: core/modules/qana/PostPlugin.cc [hunk 3]
```diff
@@ -147,7 +147,7 @@ PostPlugin::applyPhysical(QueryPlugin::Plan& p,
         if(_orderBy) {
             // Remove orderby from parallel
             // (no need to sort until we have all the results)
-            SelectStmtList::iterator i,e;
+            SelectStmtVector::iterator i,e;
```
Comment (andy-slac) [inline core/modules/qana/PostPlugin.cc]: Space missing after comma.

File: core/modules/qana/PostPlugin.cc [hunk 4]
```diff
@@ -158,7 +158,7 @@ PostPlugin::applyPhysical(QueryPlugin::Plan& p,
         LOGF_INFO("Query is non-chunked");
         // Make sure orderby is in the "parallel" section (which is not
         // really parallel). No merge is needed.
-        SelectStmtList::iterator i,e;
+        SelectStmtVector::iterator i,e;
```
Comment (andy-slac) [inline core/modules/qana/PostPlugin.cc]: Same here.

File: core/modules/query/BoolTerm.cc [hunk 5]
```diff
@@ -219,12 +219,13 @@ bool BoolFactor::_checkParen(BfTerm::PtrList& terms) {
...
 boost::shared_ptr<BoolTerm> BoolFactor::getReduced() {
     // Get reduced versions of my children.
-    BfTerm::PtrList newTerms;
+    BfTerm::PtrVector newTerms;
     bool hasReduction = false;
     hasReduction = _reduceTerms(newTerms, _terms);
     // Parentheses reduction
     if(_checkParen(newTerms)) {
-        newTerms.pop_front();
+        assert(!newTerms.empty());
```
Comment (andy-slac) [inline core/modules/query/BoolTerm.cc]: I do not think that assert will do anything useful here, `_checkParen` already guarantees what you need.

File: core/modules/query/BoolTerm.h [hunk 6]
```diff
@@ -33,8 +33,6 @@
   */
 
 // System headers
-#include <list>
-#include <stack>
```
Comment (andy-slac) [inline core/modules/query/BoolTerm.h]: include vector header

File: core/modules/query/QueryContext.h [hunk 7]
```diff
@@ -29,7 +29,6 @@
   */
 
 // System headers
-#include <list>
```
Comment (andy-slac) [inline core/modules/query/QueryContext.h]: include vector header

File: core/modules/query/ValueExpr.h [hunk 8]
```diff
@@ -30,7 +30,6 @@
 
 // System headers
 #include <iostream>
-#include <list>
```
Comment (andy-slac) [inline core/modules/query/ValueExpr.h]: add vector include

File: core/modules/query/testRepr.cc [hunk 9]
```diff
@@ -84,22 +84,24 @@ const std::string RenderedBoolTermFromRPN(const char **rpn)
             AndTerm::Ptr andt = boost::make_shared<AndTerm>();
             for(int i=0; i<opcount; ++i) {
                 andt->_terms.push_back(pdl.front());
-                pdl.pop_front();
+                assert(!pdl.empty());
+                pdl.erase(pdl.begin());
             }
-            pdl.push_front(andt);
+            pdl.insert(pdl.begin(),andt);
```
Comment (andy-slac) [inline core/modules/query/testRepr.cc]: space after comma

File: core/modules/query/testRepr.cc [hunk 10]
```diff
@@ -84,22 +84,24 @@ const std::string RenderedBoolTermFromRPN(const char **rpn)
...
         } else if (!strcmp(*t, "OR")) {
             OrTerm::Ptr ort = boost::make_shared<OrTerm>();
             for(int i=0; i<opcount; ++i) {
                 ort->_terms.push_back(pdl.front());
-                pdl.pop_front();
+                assert(!pdl.empty());
```
Comment (andy-slac) [inline core/modules/query/testRepr.cc]: assert is not needed at this point.

File: core/modules/query/types.h [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+/**
+ * @file
+ *
+ * @ingroup global
+ *
+ * @brief Query-related C++ types used by query parser and qana module
+<<<<<<< HEAD
```
Comment (andy-slac) [inline core/modules/query/types.h]: This looks like leftover stuff from conflict resolution

**Comment by fritzm** [comment]
Merged Feb 20, 2015 -- SHA ecd5c2a

**Pull request closed without merging**